diff -Naur gcc-4.5.3-orig/gcc/config/i386/linux64.h gcc-4.5.3/gcc/config/i386/linux64.h
--- gcc-4.5.3-orig/gcc/config/i386/linux64.h	2010-03-24 21:44:48.000000000 +0100
+++ gcc-4.5.3/gcc/config/i386/linux64.h	2011-06-07 18:42:31.782999161 +0200
@@ -59,7 +59,7 @@
    done.  */
 
 #define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/lib/ld-linux-x86-64.so.2"
 
 #if TARGET_64BIT_DEFAULT
 #define SPEC_32 "m32"
diff -Naur gcc-4.5.3-orig/gcc/config/i386/t-linux64 gcc-4.5.3/gcc/config/i386/t-linux64
--- gcc-4.5.3-orig/gcc/config/i386/t-linux64	2009-04-21 21:03:23.000000000 +0200
+++ gcc-4.5.3/gcc/config/i386/t-linux64	2011-06-07 18:44:21.501158404 +0200
@@ -25,7 +25,7 @@
 
 MULTILIB_OPTIONS = m64/m32
 MULTILIB_DIRNAMES = 64 32 
-MULTILIB_OSDIRNAMES = ../lib64 $(if $(wildcard $(shell echo $(SYSTEM_HEADER_DIR))/../../usr/lib32),../lib32,../lib)
+MULTILIB_OSDIRNAMES = ../lib ../lib32
 
 LIBGCC = stmp-multilib
 INSTALL_LIBGCC = install-multilib
