*** a/steam.sh	Tue Nov 17 21:07:52 2015
--- b/steam.sh	Thu Dec 24 12:40:13 2015
***************
*** 55,61 ****
  	--error)
  		title=$"Error"
  		;;
! 	--warning)
  		title=$"Warning"
  		;;
  	*)
--- 55,61 ----
  	--error)
  		title=$"Error"
  		;;
! 	--sorry)
  		title=$"Warning"
  		;;
  	*)
***************
*** 67,73 ****
  	echo -e "$title: $*"
  
  	if [ -z "$STEAMOS" ]; then
! 		if ! zenity "$style" --text="$*" 2>/dev/null; then
  			# Save the prompt in a temporary file because it can have newlines in it
  			tmpfile="$(mktemp || echo "/tmp/steam_message.txt")"
  			echo -e "$*" >"$tmpfile"
--- 67,73 ----
  	echo -e "$title: $*"
  
  	if [ -z "$STEAMOS" ]; then
! 		if ! kdialog "$style" "$*" 2>/dev/null; then
  			# Save the prompt in a temporary file because it can have newlines in it
  			tmpfile="$(mktemp || echo "/tmp/steam_message.txt")"
  			echo -e "$*" >"$tmpfile"
***************
*** 97,108 ****
  			answer=accepted
  		else
  			answer=declined
! 			output=$(zenity --width 650 --height 500 --text-info --title=$"Steam Install Agreement" --filename="$LICENSE" --checkbox=$"I have read and accept these terms." 2>&1)
  			STATUS=$?
! 			if echo $output | grep "status 1:" >/dev/null; then
! 				# Zenity couldn't launch a window
! 				STATUS=-1
! 			fi
  			case $STATUS in
  			0)	# The agreement was accepted
  				answer=accepted
--- 97,105 ----
  			answer=accepted
  		else
  			answer=declined
! 			
  			STATUS=$?
! 			kdialog --textbox $LICENSE 650 500 --title=$"Steam Install Agreement"
  			case $STATUS in
  			0)	# The agreement was accepted
  				answer=accepted
